numpy/tests/test_public_api.py

```diff
@@ -276,6 +276,7 @@ def test_NPY_NO_EXPORT():
     "lib.utils",
     "linalg.lapack_lite",
     "linalg.linalg",
+    "hook-numpy",
```

Comment:
Must it have this exact name, or can you prepend an underscore?

Reply:
It must have that name but we can put it in a private underscore-prefixed subdirectory if you prefer? I'd have though that the `-` making it an invalid submodule would be sufficient to keep users and IDEs from mistaking it as public API but I was quite surprised that `pkgutil.walk_packages()` treated it as a submodule rather than just a data file.

Reply:
I'm going to do it anyway. It's clear that the test suite doesn't like it - even with this line here.

Reply:
Thanks. Yes that's the right thing to do here. The API is enough of a mess as it is, and we've tried hard to add underscores to all new files. Users don't read docs, so if `import numpy.hook_numpy` works then it looks too public.

Reply:
I'm going to stick it in a `numpy._pyinstaller` submodule. Shall I put the test in there too?

Reply:
That sounds like a good plan.